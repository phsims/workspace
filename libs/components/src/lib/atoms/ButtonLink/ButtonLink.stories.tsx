import type { Meta, StoryObj } from '@storybook/react';
import { ButtonLink } from './ButtonLink';

import { within } from '@storybook/testing-library';
import { expect } from '@storybook/jest';

const meta: Meta<typeof ButtonLink> = {
  component: ButtonLink,
  title: 'ButtonLink',
};
export default meta;
type Story = StoryObj<typeof ButtonLink>;

export const Primary = {
  args: {
    text: 'Test Button',
    link: '/test',
  },
};

export const Heading: Story = {
  args: { ...Primary.args },
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    expect(canvas.getByText(/Welcome to ButtonLink!/gi)).toBeTruthy();
  },
};
