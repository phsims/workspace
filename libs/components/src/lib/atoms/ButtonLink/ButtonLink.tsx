import { Button, Link } from '@mui/material';


export interface ButtonLinkProps {
  text: string;
  link:string,
  variant?: 'text' | 'outlined' | 'contained';
  color?:
    | 'primary'
    | 'inherit'
    | 'secondary'
    | 'success'
    | 'error'
    | 'info'
    | 'warning';
}

export function ButtonLink({
  text,
  link,
  variant='contained',
  color = 'primary',
}: ButtonLinkProps) {
  return (
    <Link href={link}>
      <Button variant={variant} color={color}>
        {text}
      </Button>
    </Link>
  );
}

export default ButtonLink;