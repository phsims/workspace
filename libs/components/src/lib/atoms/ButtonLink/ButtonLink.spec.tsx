
import { render } from '@testing-library/react';
import ButtonLink from './ButtonLink';

describe('ButtonLink component', () => {
  it('should match snapshot', () => {
    const { baseElement } = render(
      <ButtonLink text="Test Button" link="/test" />
    );
    expect(baseElement).toMatchSnapshot();
  });
  it('renders button with text and link', () => {
    const { getByRole } = render(
      <ButtonLink text="Test Button" link="/test" />
    );

    const linkElement = getByRole('link');
    expect(linkElement).toBeDefined();
    expect(linkElement).toHaveAttribute('href', '/test');
  });

  it('renders button with specified variant and color', () => {
    const { getByRole } = render(
      <ButtonLink
        text="Test Button"
        link="/test"
        variant="outlined"
        color="secondary"
      />
    );

    const buttonElement = getByRole('button', { name: 'Test Button' });
    expect(buttonElement).toHaveClass('MuiButton-outlined');
    expect(buttonElement).toHaveClass('MuiButton-colorSecondary');
  });

  it('renders button with default variant and color if not provided', () => {
    const { getByRole } = render(
      <ButtonLink text="Test Button" link="/test" />
    );

    const buttonElement = getByRole('button', { name: 'Test Button' });
    expect(buttonElement).toHaveClass('MuiButton-contained');
    expect(buttonElement).toHaveClass('MuiButton-colorPrimary');
  });
});
