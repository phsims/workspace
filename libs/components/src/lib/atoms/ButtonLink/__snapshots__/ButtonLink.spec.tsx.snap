// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonLink component should match snapshot 1`] = `
<body>
  <div>
    <a
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1xylxj1-MuiTypography-root-MuiLink-root"
      href="/test"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-sghohy-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Test Button
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </a>
  </div>
</body>
`;
